src/libsystemd-network/sd-dhcp-lease.c

```diff
@@ -444,27 +442,31 @@ static int lease_parse_search_domains(const uint8_t *option, size_t len, char **
                                         return r;
 
                                 n += r;
-                                continue;
                         } else if ((c & 0xc0) == 0xc0) {
                                 /* Pointer */
 
                                 uint8_t d;
                                 uint16_t ptr;
+
+                                if (pos >= len)
+                                        return -EMSGSIZE;
+
                                 d = option[pos++];
                                 ptr = (uint16_t) (c & ~0xc0) << 8 | (uint16_t) d;
 
-                                /* Jumps are limited to a prior occurrence of a complete domain (labels ending with 0),
-                                 * according to RFC1035 4.1.4. */
+                                /* Jumps are limited to a "prior occurrence" (RFC-1035 4.1.4) */
```

Comment:
That would allow loops. For example, assuming we have this in the option data,

3 F O O ptr(points to 0)

If we allow ptr, we would end up in a infinite loop, wouldn't we?